configs/components/facter.rb

```diff
@@ -112,7 +112,7 @@
     skip_blkid = 'OFF'
   elsif platform.is_rpm?
     if (platform.is_el? && platform.os_version.to_i >= 6) || (platform.is_sles? && platform.os_version.to_i >= 11) || platform.is_fedora?
-      pkg.build_requires "libblkid-devel"
+      pkg.build_requires "libblkid-devel" unless platform.architecture == "s390x"
```

Comment:
`||` and `or` is basic precedence; how will the conditional be evaluated from left to right? But as a rule, always use `||`. It's consistent and predictable.